fs-storage/src/monoid.rs

```diff
@@ -0,0 +1,17 @@
+// Currently, we have three structures: Tags (HashSet), Properties (HashSet), Score (int).
+// In fact, HashSet already implements a union function,
+// so only a special function for integers is needed.
+// CRDTs can be considered later when we need to add structures that require
+// more powerful combine semantics.
+
+pub trait Monoid<V> {
+    fn neutral() -> V;
```

Comment:
same doubt
cc @twitu

Reply:
There's no practical difference as such. It's just that Monoid is defined as having a combine and an empty operator. This is consistent across all language implementations (for e.g. [haskell](https://hackage.haskell.org/package/base-4.19.1.0/docs/Data-Monoid.html)). It's better to be consistent with the definition of a monoid. Otherwise, it might be better to call it something else.

Reply:
Yep, `neutral` is from standard Monoid definitions. It's coming from algebra, where it is defined over some operation `+` together with some set `X`. Pair (`+`, `X`) must possess 2 properties to be considered a Monoid:
1. `X` contains a `neutral` element, i.e. such `0` that `0 + a = a + 0 = 0` for any `a`.
2. `+` is associative, i.e. `a + (b + c)` is same as `(a + b) + c`.

It's the simplest and the easiest to use algebra abstraction :) It helps to define something what can be summed. E.g. strings, vectors, sets, etc. Defining it as a trait helps to build merge pattern, abstract from particular data.